src/webots/control/WbController.cpp

```diff
@@ -579,18 +573,24 @@ void WbController::appendMessageToBuffer(const QString &message, QString *buffer
 void WbController::flushBuffer(QString *buffer) {
   // Split string into lines by detecting '\n', then send lines one by one to WbLog.
   // When several streams or several controllers are used, this prevents to mix unrelated lines
-  if ((*buffer)[0] == '\f') {
-    WbLog::clear();
-    buffer->remove(0, 1);
-  }
   int index = buffer->indexOf('\n');
   while (index != -1) {
-    // extract line and prepend "[controller_name] "
-    QString line = mPrefix + buffer->mid(0, index + 1);
+    QString ansiString;
+    int count = 0;
+    int l = buffer->length();
+    while (buffer->at(count) == '\x1b') {
+      int start = count;
+      count += 2;  // skipping '['
+      while (count < l && buffer->at(count++) < '\x40')
+        ;
+      ansiString += buffer->mid(start, count - start);
+    }
+    // html formatted string ex: "<span style=\"color:#66bb6a;\">Some Text\n</span>"
+    const QString line = ansiString + buffer->mid(count, index + 1 - count);
```

Comment:
Here is what I mean:
```suggestion
    const QString line = buffer->mid(0, index + 1);
```

Reply:
Oh, I'm sorry I thought you mentioned the part below, I didn't try without that part since I thought that was your commit on this branch, I can still try if you want.